skxray/recip.py

```diff
@@ -194,3 +194,106 @@ def hkl_to_q(hkl_arr):
     """
 
     return np.linalg.norm(hkl_arr, axis=1)
+
+
+def q_rings(num_qs, first_q, delta_q, q_val, step_q=None):
+    """
+    This will find the indices of the required Q rings, find the bin
+    edges of the Q rings, and count the number of pixels in each Q ring.
+
+    Parameters
+    ----------
+    num_qs : int
+        number of Q rings
+
+    first_q : float
+        Q value of the first Q ring
+
+    delta_q : float
+        thickness of the Q ring
+
+    q_val : ndarray
+        Q space values for each pixel in the detector
+        shape is ([detector_size[0]*detector_size[1]], ) or
+        ([detector_size[0]*detector_size[1]], 1)
+
+    step_q : float, optional
+        step value for the next Q ring from the end of the previous Q ring
+
+    Returns
+    -------
+    q_inds : ndarray
+        indices of the Q values for the required rings
+
+    q_ring_val : ndarray
+        edge values of each Q ring
+        shape is (num_qs, 2)
+
+    num_pixels : ndarray
+        number of pixels in each Q ring
+    """
+
+    if (delta_q < 0):
```

Comment:
Python doesn't need parenthesis for if statement. If you don't mind, please modify.